dashboard/app/helpers/users_helper.rb

```diff
@@ -199,99 +199,95 @@ def script_progress_for_users(users, script)
+        )
 
-        progress[level_id] = level_progress
+        next unless level_progress
 
-        # Just in case this level has multiple pages, in which case we add an additional
+        # if the level has multiple pages, we add an additional
         # array of booleans indicating which pages have been completed.
-        pages_completed = get_pages_completed(user, sl)
-
-        next unless pages_completed
+        level_progress[:pages_completed] = get_pages_completed(user, sl)
 
-        progress[level_id][:pages_completed] = pages_completed
-        pages_completed.each_with_index do |result, index|
-          progress["#{level_id}_#{index}"] = {
-            result: result,
-            submitted: level_progress[:submitted],
-            readonly_answers: level_progress[:readonly_answers]
-          }.compact
-        end
+        progress[level_id] = level_progress.compact
       end
     end
     progress
   end
 
   # Summarizes a user's progress on a particular level
-  private def get_level_progress(user_level, script_level, paired_user_levels, include_timestamp)
-    completion_status = activity_css_class(user_level)
-    locked = user_level&.show_as_locked?(script_level.lesson) || script_level.lesson.lockable? && !user_level
-
-    # For levels that are not tried we will either return no progress or a locked status for locked levels.
-    # In either case, we do not need a full set of progress data so we return early here.
-    if completion_status == LEVEL_STATUS.not_tried
-      return {
-        status: completion_status,
-        locked: locked || nil
-      }
+  private def get_level_progress(
+    user_level,
+    script_level,
+    paired_user_levels,
+    include_timestamp
+  )
+    # if we don't have a user level, we don't need to return anything,
+    # unless the lesson is lockable, in which case that means it's locked
+    if user_level.nil?
+      if script_level.lesson.lockable?
+        return {locked: true}
+      else
+        return nil
+      end
     end
 
-    # a user_level is submitted if the state is submitted UNLESS it is a peer reviewable level that has been reviewed
-    submitted = !!user_level&.submitted &&
-      !(user_level.level.try(:peer_reviewable?) && [ActivityConstants::REVIEW_REJECTED_RESULT, ActivityConstants::REVIEW_ACCEPTED_RESULT].include?(user_level.best_result))
-
     return {
-      status: completion_status,
-      locked: locked || nil,
-      result: user_level&.best_result || 0,
-      submitted: submitted || nil,
-      readonly_answers: !!user_level&.readonly_answers || nil,
-      paired: (paired_user_levels.include? user_level&.id) || nil,
-      last_progress_at: include_timestamp ? user_level&.updated_at&.to_i : nil,
-      time_spent: user_level&.time_spent&.to_i
+      status: activity_css_class(user_level),
+      locked: user_level.show_as_locked?(script_level.lesson) || nil,
+      result: user_level.best_result || nil,
```

Comment:
Just curious, do we still use both `status` and `result` on the front-end?  (If we're close to eliminating one or the other, let's tackle in future PR.)

Reply:
we still use both. it will be great to someday get rid of one, but that's definitely out of scope for now.